dashboard/test/models/script_test.rb

```diff
@@ -194,6 +194,52 @@ def populate_cache_and_disconnect_db
     assert_equal 'Stage1', second.name
   end
 
+  test 'all fields can be set and then removed on reseed' do
+    # First, seed using a .script file that sets something truthy for all properties.
+    script_file_all_properties = File.join(self.class.fixture_path, 'test-all-fields.script')
+    scripts, _ = Script.setup([script_file_all_properties])
+    script = scripts.first
+
+    # Not testing new_name since it causes a new script to be created.
+    assert_equal false, script.hidden? # defaults to true, so we want to verify it was explicitly set to false
+    assert script.login_required?
+    assert_equal 'csd1', script.family_name
+
+    # Seed using an empty .script file with the same name. Verify that this sets all values back to defaults.
+    script_file_no_properties = File.join(self.class.fixture_path, 'duplicate_scripts', 'test-all-fields.script')
+    Script.setup([script_file_no_properties])
+    script.reload
+
+    assert script.hidden? # defaults to true
+    assert_equal false, script.login_required?
+    assert_nil script.family_name
+  end
+
+  test 'all properties can be set and then removed on reseed' do
+    # First, seed using a .script file that sets something truthy for all properties.
+    script_file_all_properties = File.join(self.class.fixture_path, 'test-all-properties.script')
+    scripts, _ = Script.setup([script_file_all_properties])
+    script = scripts.first
+
+    assert_equal 20, script.properties.keys.length
+    script.properties.values.each {|v| assert v}
+
+    # Seed using an empty .script file with the same name. Verify that this sets all values back to defaults.
+    script_file_no_properties = File.join(self.class.fixture_path, 'duplicate_scripts', 'test-all-properties.script')
+    Script.setup([script_file_no_properties])
+    script.reload
+
+    # The values below do not get reset to defaults. TODO: determine if this behavior should be changed.
```

Comment:
behavior should be changed for all of these, it is just a question of verifying that this won't break anything first.

Reply:
I guess the word "should" is a bit overloaded :-) the correct behavior is definitely to fix all of these. I'm suggesting that we do this work now, since incorrect values for any of these fields would lead to undesirable behavior on our web site. but I am open to pushback